### PR TITLE
Tickets/dm 36063

### PR DESCRIPTION
**Pull request by pcortes80**
Please help to check this PR. Thanks.

### CODE REVIEW

File: doc/userManual.md [hunk 1]
```diff
@@ -195,7 +195,13 @@ If not, there should be something wrong in the **TcpServer**.
 
 The `TcpServer` module uses the `GenRead` user event to send the explanation of the errors that occur internally.
 The error is cleared by the module and the user has to manage it.
-The error explanation comes from the error LabVIEW internal database.
+
+The **TcpServerBase.clearErrorAndSendExplanation.vi** sends the generic user event string with the error code and the source of the error.
+It was not possible to get the `Possible reason(s)` of error, because the vis that get this information are **Simple Error Handler.vi** and **General Error Handler.vi**, but they have internally two vis that are not supported in real-time targets.
+We tried to find a solution on NI forums without success.
+We went deep inside these mentioned LabVIEW functions, and found how LabVIEW searches into an internal error database, using a private function called **Error Code Database.vi** as part of the **error.llb**.
+However, we did not use this vi because is not a public vi and a non-maintainable vi.
```
Comment (teweitsai) [inline doc/userManual.md:203]: because "it" is not

File: doc/userManual.md [hunk 2]
```diff
@@ -195,7 +195,13 @@ If not, there should be something wrong in the **TcpServer**.
 
 The `TcpServer` module uses the `GenRead` user event to send the explanation of the errors that occur internally.
 The error is cleared by the module and the user has to manage it.
-The error explanation comes from the error LabVIEW internal database.
+
+The **TcpServerBase.clearErrorAndSendExplanation.vi** sends the generic user event string with the error code and the source of the error.
```
Comment (teweitsai) [inline doc/userManual.md:199]: Please help to put the link of NI error code list that the user knows where to look for the error details:
https://www.ni.com/docs/en-US/bundle/labview/page/lverror/misc_lv_error_codes.html

Maybe there are more. Please help to double check.

File: doc/versionHistory.md [hunk 3]
```diff
@@ -1,5 +1,12 @@
 # Version History
 
+0.4.2
+
+- Replace **General Error Handler.vi** with **Unbundle By Name** function to create the string with error code and error explanation in **TcpServerBase.clearErrorAndSendExplnation.vi.**
```
Comment (teweitsai) [inline doc/versionHistory.md:5]: error code and error explanation --> error code and error source

File: simulator/README.md [hunk 4]
```diff
@@ -133,3 +133,18 @@ Note:
...
+If you want to use this vi as a top-level vi and want to compile it in a real-time target, avoid using property nodes.
+Since in real-time target there is no front panel, no value will be written to the front panel controls and indicators, and this can cause unexpected vi behavior, as is explained in [this document](https://knowledge.ni.com/KnowledgeArticleDetails?id=kA00Z000000kGsVSAU&l=en-US).
+- If you want to use this simulator in a real-time target, create the build specification, go to the properties, and follow this:
+    - Go to `Source Files` category and add the `default.ini` file in the `Always Included` box.
+    - Go to `Destinations` category and click the blue cross (+) under the `Destination` box to create a new directory.
+    Name it `config` and modify the path in the `Destination path` box as `/home/lvuser/natinst/config`.
+    - Go to `Source File Settings` category, click the `default.ini` file in the `Porject Files` box, and select `config` in the `Destination` combo box.
+    - Go to `Preview` category and click `Generate Preview`.
+    The preview files structure has to be like the following image:
```
Comment (teweitsai) [inline simulator/README.md:148]: The preview should look like the following:

**Review by teweitsai — APPROVED** [review]
Looks good. Thanks for this effort!

File: doc/userManual.md (hunk 1 above)
Comment (pcortes80) [reply doc/userManual.md:203]: Done

File: doc/userManual.md (hunk 2 above)
Comment (pcortes80) [reply doc/userManual.md:199]: Done.

File: simulator/README.md (hunk 4 above)
Comment (pcortes80) [reply simulator/README.md:148]: Fixed.

File: doc/versionHistory.md (hunk 3 above)
Comment (pcortes80) [reply doc/versionHistory.md:5]: Fixed.

**Pull request merged**